hdat/suite.py

```diff
@@ -97,3 +97,79 @@ def _collect_suite_classes(directory):
                     if hdat_suite_class in inspect.getmro(value) and hdat_suite_class != value:
                         suite_classes.append(value)
     return suite_classes
+
+
+class MetricsChecker:
+    '''
+    Helper class for comparing metrics of new results vs golden results in suite.check()
+    '''
+    def __init__(self, old, new):
+        self._old = old
+        self._new = new
+        self._match = True
+        self._msgs = []
+        if old.keys() != new.keys():
+            msg = 'Metric(s) in golden result {} were missing from the new run'
```

Comment:
I just thought of this, but what if metrics are taken away on the new run?

Reply:
If they're taken away, they'll get this message but the rest of the checks will still run. But if they're added, they get the same message, which is confusing.

Reply:
Yeah, maybe we can show a different message depending on whether keys were added vs removed?